src/scripts/main.js

```diff
@@ -1,3 +1,261 @@
+}
+
+function slideHorizontally(direction) {
+  for (let row = 0; row < rows; row++) {
+    const newRow = board[row].filter(cell => cell !== 0);
+
+    if (newRow.length !== 4) {
+      if (direction === 'right') {
+        do {
+          newRow.unshift(0);
+        } while (newRow.length < 4);
+      } else {
+        do {
+          newRow.push(0);
+        } while (newRow.length < 4);
+      }
+    }
+    board.push(newRow);
+  }
+
+  board = board.slice(-4);
+
+  toMerge(direction);
+  check();
+
+  toUpdateBoard();
+}
+
+function slideVertically(direction) {
+  for (let row = 0; row < rows; row++) {
+    const newRow = [];
+
+    for (let cell = 0; cell < cells; cell++) {
+      newRow.push(board[cell][row]);
+    }
+
+    const newClmnDirection = newRow.filter(elem => elem !== 0);
+
+    if (newClmnDirection.length !== cells) {
+      if (direction === 'down') {
+        do {
+          newClmnDirection.unshift(0);
+        } while (newClmnDirection.length < cells);
+      } else {
+        do {
+          newClmnDirection.push(0);
+        } while (newClmnDirection.length < cells);
+      }
+    }
+
+    for (let cell = 0; cell < cells; cell++) {
+      board[cell][row] = newClmnDirection[cell];
+    }
+  }
+
+  toMerge(direction);
+  check();
+
+  toUpdateBoard();
+}
+
+function toUpdateBoard() {
+  if (JSON.stringify(tempBoard) !== JSON.stringify(board)) {
+    getRandomCell();
+  }
+
+  render();
+}
+
+function toUpdateCell([row, cell], [cellMerged, rowMerged]) {
+  tempBoard = JSON.parse(JSON.stringify(board));
+
+  board[row][cell] *= 2;
+  board[rowMerged][cellMerged] = 0;
+
+  score += board[row][cell];
+}
+
+function toMerge(direction) {
+  if (direction === 'left' || direction === 'right') {
+    board.forEach((row, rowIndex) => {
+      row.forEach((cell, cellIndex) => {
+        if (direction === 'left' && cell === row[cellIndex + 1] && cell > 0) {
+          toUpdateCell([rowIndex, cellIndex], [cellIndex + 1, rowIndex]);
+        }
+
+        if (direction === 'right' && cell === row[cellIndex - 1] && cell > 0) {
+          toUpdateCell([rowIndex, cellIndex], [cellIndex - 1, rowIndex]);
+        }
+      });
+    });
+  }
+
+  if (direction === 'down' || direction === 'up') {
+    if (direction === 'down') {
+      for (let row = rows - 1; row > 0; row--) {
+        for (let cell = 0; cell < cells; cell++) {
+          const currentCell = board[row][cell];
+
+          if (currentCell === board[row - 1][cell] && currentCell > 0) {
+            toUpdateCell([row, cell], [cell, row - 1]);
+          }
+        }
+      }
+    }
+
+    if (direction === 'up') {
+      for (let row = 0; row < rows - 1; row++) {
+        for (let cell = 0; cell < cells; cell++) {
+          const currentCell = board[row][cell];
+
+          if (currentCell === board[row + 1][cell] && currentCell > 0) {
+            toUpdateCell([row, cell], [cell, row + 1]);
+          }
+        }
+      }
+    }
+  }
+}
+
+function render() {
```

Comment:
Consider using more descriptive function names instead of `render`. For example, `updateGameBoard`. 🤔